### PR TITLE
Use `gcov-viewer` for coverage highlighting

### PR DESCRIPTION
**Pull request by emilienlemaire**

### CODE REVIEW

File: src/lsp/superbol_free_lib/vscode_extension.ml [hunk 1]
```diff
@@ -546,3 +558,6 @@ let manifest =
       "workspace";                                 (* <- run on the workspace *)
     ]
     ~contributes
+    ~extensionDependencies: [
+      "JacquesLucke.gcov-viewer";
```
Comment (ddeclerck) [inline src/lsp/superbol_free_lib/vscode_extension.ml:562]: Is this is hard dependency or an optional one ?

Comment (emilienlemaire) [reply src/lsp/superbol_free_lib/vscode_extension.ml:562]: Hard, I can make it optional, but I don't know how well the execution of the commands would work if the extension is missing

File: README.md [hunk 2]
```diff
@@ -228,12 +228,17 @@ Coverage data can be shown after the execution of a program that was
...
 display coverage on a line-by-line basis, by highlighting the lines of
 your source code using colors that represent their coverage status.
+To enable coverage highlighting, you can open the command palette
+(<kbd>Ctrl+Shift+P</kbd>), and select `SuperBOL: Show program coverage`.
+You can also hide the highlighting with the command
+`SuperBOL: Hide program coverage`, and update it after re-executing your
+program with `SuperBOL: Update program coverage`.
 
-> [!NOTE]
+<!-- > [!NOTE]
```
Comment (nberth) [inline README.md:237]: I think you can just remove those lines instead of commenting them out.

File: README.md [hunk 3]
```diff
@@ -228,12 +228,17 @@ Coverage data can be shown after the execution of a program that was
 compiled to generate this information terminates.  SuperBOL will
 display coverage on a line-by-line basis, by highlighting the lines of
 your source code using colors that represent their coverage status.
+To enable coverage highlighting, you can open the command palette
+(<kbd>Ctrl+Shift+P</kbd>), and select `SuperBOL: Show program coverage`.
```
Comment (nberth) [inline README.md:232]:
```suggestion
(or type <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>P</kbd>), and select `SuperBOL: Show program coverage`.
```

Comment (nberth) [reply README.md:232]: We typeset commands as English titles: with full capitals. So here that should be `SuperBOL: Show Program Coverage` (`SuperBOL: Show Coverage` would be better I think, "program" may not be relevant when working on at module level).

(Same for the other commands below)

File: src/lsp/superbol_free_lib/vscode_extension.ml [hunk 4]
```diff
@@ -520,6 +520,18 @@ let contributes =
         ~title:"Write Project Configuration"
         ~category:"SuperBOL"
       (* ~enablement:"!inDebugMode" *);
+      Manifest.command ()
+        ~command:"superbol.coverage.show"
+        ~title:"Show program coverage"
```
Comment (nberth) [inline src/lsp/superbol_free_lib/vscode_extension.ml:525]:
```suggestion
        ~title:"Show Coverage"
```

File: src/lsp/superbol_free_lib/vscode_extension.ml [hunk 5]
```diff
@@ -520,6 +520,18 @@ let contributes =
...
+        ~category:"SuperBOL";
+      Manifest.command ()
+        ~command:"superbol.coverage.hide"
+        ~title:"Hide program coverage"
```
Comment (nberth) [inline src/lsp/superbol_free_lib/vscode_extension.ml:529]:
```suggestion
        ~title:"Hide Coverage"
```

File: src/lsp/superbol_free_lib/vscode_extension.ml [hunk 6]
```diff
@@ -520,6 +520,18 @@ let contributes =
...
+        ~category:"SuperBOL";
+      Manifest.command ()
+        ~command:"superbol.coverage.reload"
+        ~title:"Update program coverage"
```
Comment (nberth) [inline src/lsp/superbol_free_lib/vscode_extension.ml:533]:
```suggestion
        ~title:"Update Coverage"
```

File: src/lsp/superbol_free_lib/vscode_extension.ml (hunk 1 above)
Comment (nberth) [reply src/lsp/superbol_free_lib/vscode_extension.ml:562]: I'd go with a hard dependency as availability seems universal, at least until we have a case where this is an issue.

**Review by nberth — APPROVED** [review]
With minor typesetting suggestions.

**Comment by nberth** [comment]
I forgot: can you add an entry into `CHANGELOG.md`?
```
### Added
- Commands to show/hide/update coverage by highlighting the source code [#285](https://github.com/OCamlPro/superbol-studio-oss/pull/285)
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -4,6 +4,7 @@
 
 ### Added
 - Ability to build modules instead of executable programs [#284](https://github.com/OCamlPro/superbol-studio-oss/pull/284)
+- Commands to show/hide/update coverage by highlighting the source code, using `gcov-viewer` [#285](https://github.com/OCamlPro/superbol-studio-oss/pull/285)
```
Comment (nberth) [inline CHANGELOG.md:7]: Please swap those two entries (we sort them the other way round).

**Pull request merged**